core/src/main/java/org/infinispan/marshall/core/InternalClassIds.java

```diff
@@ -0,0 +1,21 @@
+package org.infinispan.marshall.core;
+
+/**
+ * Identifiers for marshalling class name itself; this comes handy when marshalling arrays
+ * and the array component type is interface (and therefore it does not have its AdvancedExternalizer).
+ *
+ * This identifiers don't clash with {@link org.infinispan.commons.marshall.Ids} and therefore it can use the same range.
+ */
+public interface InternalClassIds {
```

Comment:
I find the name confusing, shouldn't it be better `ClassNameIds`? Or if only using it for arrays, `ArrayClassIds`?

Why an interface? And why public? A default package final class with no constructor would be much better option for defining constants only used in `marshall.core` package.